.buildkite/x-pack/pipeline.xpack.heartbeat.yml

```diff
@@ -25,39 +25,189 @@ env:
   #Deps
   ASDF_MAGE_VERSION: 1.15.0
 
+notify:
+  - github_commit_status:
+      context: "$BEATS_PROJECT_NAME"
+
 steps:
+  - group: "Mandatory Tests"
+    key: "mandatory-tests"
+    steps:
+      - label: ":linux: Ubuntu Unit Tests"
+        key: "mandatory-linux-unit-test"
+        command: "cd $BEATS_PROJECT_NAME && mage build unitTest"
```

Comment:
we are making things explicit in the other PRs (see also https://github.com/elastic/beats/pull/38871), could you see if you can search/replace `$BEATS_PROJECT_NAME` with `x-pack/heartbeat" in the entire PR?

If we remove all occurrences of `BEATS_PROJECT_NAME` from our PRs, we'll be able to sunset a lot of common scripts.

Before attempting to replace this, please take a look my comment below about the execute command on macOS; if it works, then you can proceed with replacing this env var too.